src/typelib.erl

```diff
@@ -92,8 +92,6 @@ remove_pos({Type, _, Value})
 remove_pos({user_type, Anno, Name, Params}) when is_list(Params) ->
     {user_type, anno_keep_only_filename(Anno), Name,
      lists:map(fun remove_pos/1, Params)};
-remove_pos({type, Anno, record, Params = [{atom, AtomAnno, Name}]}) ->
-    {type, anno_keep_only_filename(Anno), record, [{atom, anno_keep_only_filename(AtomAnno), Name}]};
```

Comment:
Because this will take care of it https://github.com/josefs/Gradualizer/blob/master/src/typelib.erl#L108-L109

```erlang
remove_pos({type, _, Type, Params}) when is_list(Params) ->
    {type, erl_anno:new(0), Type, lists:map(fun remove_pos/1, Params)};
```

This also does `field_type` for free.

Reply:
But no, there is a difference. For records, `anno_keep_only_filename/1` should be called on the Anno instead of `erl_anno:new(0)`, because we need to remember which module the record came from. (A remote record comes from a remote type which expands to a record type.)

Reply:
One point where I'm not sure about this. Let's say you have a record from a header file. Right now this PR support this case

```erlang
% file.hrl
-record(record, { field :: integer() | undefined }).
```
```erlang
% file.erl
-include("file.hrl").

-spec safe(#record{ field :: integer() }) -> integer().
safe(#record{ field = I }) -> I.

-spec unsafe(#record{}) -> integer().
unsafe(#record{ field = undefined }) -> 0;
unsafe(R) -> safe(R).
```

This use case is pretty useful for me because there are placed where some checks are done upstream. If we keep the filename, wouldn't that not work anymore? I tried readding this (and supporting `field_type` for fields but it makes things fail because I guess I do an explicit file check. Should we remove the filename in `remove_pos` or I can try to find who looks for the specific file check in `typechecker.erl`.

Reply:
It would be quite good if the error messages would contain the hrl file as the filename as you say, yes. I don't know if it works or not. What I remember is that I once made gradualizer_db add "module.erl" as the file when fetching records from remote modules. This happens if an exported type expands to a record defined in the other module.

I guess the only way to test it would be to check the output of a test case where the error message is referring to a record defined in another module.

Reply:
I've readded the case but kept it commented. I think we'll keep it for future work because I want to start integrating gradualizer in our codebase to try and find more bugs. I will resolve this for now.